File: gluecodium/src/test/resources/smoke/equatable/output/android/com/example/smoke/EquatableInterfaceImpl.java
/*
 *
 */
package com.example.smoke;
import com.example.NativeBase;
class EquatableInterfaceImpl extends NativeBase implements EquatableInterface {
    protected EquatableInterfaceImpl(final long nativeHandle) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @Override
    public native boolean equals(Object rhs);
    @Override
    public native int hashCode();
}

File: gluecodium/src/test/resources/smoke/equatable/output/android-kotlin/com/example/smoke/EquatableInterfaceImpl.kt
/*

 *
 */

@file:JvmName("EquatableInterfaceImpl")

package com.example.smoke

import com.example.NativeBase

/**
 * @suppress
 */
class EquatableInterfaceImpl : NativeBase, EquatableInterface {
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}




    override external fun equals(other: Any?) : Boolean
    override external fun hashCode(): Int

    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
